arden/ArdenLexer.g4

```diff
@@ -0,0 +1,344 @@
+lexer grammar ArdenLexer;
+options { caseInsensitive = true; }
+
+// Keywords
+ABS:            'abs';
+ACTION:         'action:';
+ADD:            'add';
+AFTER:          'after';
+AGGREGATE:      'aggregate';
+AGO:            'ago';
+ALL:            'all';
+AND:            'and';
+ANY:            'any';
+APPLICABILITY:  'applicability';
+ARCCOS:         'arccos';
+ARCSIN:         'arcsin';
+ARCTAN:         'arctan';
+ARDEN:          'arden:';
+ARDEN_VERSION:  'version'[ ]('2' (DOT ( '1' | '2' | '5' | '6' | '7' | '8' | '9' | '10'))? | [3]);
```

Comment:
Please don't mix string literals (`' '`) and range literals (`[ ]`) within one rule:

```suggestion
ARDEN_VERSION:  'version ' ('2' ('.' ( '1' | '2' | '5' | '6' | '7' | '8' | '9' | '10'))? | '3');
```